### PR TITLE
Add support for gpu devices reservation

### PR DESCRIPTION
**Pull request by ndeloof**
**What I did**
In addition to swarm-style for generic_resources, support devices reservation as defined by Adopt https://github.com/compose-spec/compose-spec/pull/100

/test-ecs

**(not mandatory) A picture of a cute animal, if possible in relation with what you did**

### CODE REVIEW

File: ecs/convert.go [hunk 1]
```diff
@@ -562,6 +562,11 @@ func gpuRequirements(s types.ServiceConfig) int64 {
 					return resource.DiscreteResourceSpec.Value
 				}
 			}
+			for _, device := range reservations.Devices {
+				if len(device.Capabilities) == 1 && device.Capabilities[0] == "gpu" {
```
Comment (aiordache) [inline ecs/convert.go:566]: Do we want to restrict the capabilities to "gpu" only? Could we be more flexible? I thought these flags are passed to a Docker Engine and the engine can return an error if we request a capability that is not supported.

Comment (ndeloof) [reply ecs/convert.go:566]: we don't target docker engine here but AWS EC2 and need to select machine type based on number of requested GPUs

**Review by gtardif — APPROVED** [review]
Unsure if `CheckDeployResourcesDevicesCapabilities` is unused and should be removed, otherwise LGTM

**Review by aiordache — APPROVED** [review]

**Comment by ndeloof** [comment]
CheckDeployResourcesDevicesCapabilities is invoked by compose-go https://github.com/compose-spec/compose-go/blob/d948014098818897aa3fe60cc1858c65788212d3/compatibility/deploy.go#L168

**Pull request merged**